rir/src/compiler/opt/scope_resolution.cpp

```diff
@@ -99,7 +99,9 @@ class TheScopeResolution {
                         // This load can be resolved to a unique value
                         aval.ifSingleValue([&](Value* val) {
                             if (ldfun) {
-                                replaceLdFun(val);
+                                // The force can leak the environment where the ldfun
+                                // we are replacing would be executed
+                                replaceLdFun(val, ldfun->env());
```

Comment:
Still not sure `ldfun->env()` is the correct one... We want the one where the `ldfun` instruction is executed, this gives us the one where it does the lookup. Do we know they are always the same?